ete_tests/src/EdgeCases/Interface/Character.elm

```diff
@@ -0,0 +1,52 @@
+-- Do not manually edit this file, it was auto-generated by dillonkearns/elm-graphql
+-- https://github.com/dillonkearns/elm-graphql
+
+
+module EdgeCases.Interface.Character exposing (..)
+
+import EdgeCases.InputObject
+import EdgeCases.Interface
+import EdgeCases.Object
+import EdgeCases.Scalar
+import EdgeCases.ScalarCodecs
+import EdgeCases.Union
+import Graphql.Internal.Builder.Argument as Argument exposing (Argument)
+import Graphql.Internal.Builder.Object as Object
+import Graphql.Internal.Encode as Encode exposing (Value)
+import Graphql.Operation exposing (RootMutation, RootQuery, RootSubscription)
+import Graphql.OptionalArgument exposing (OptionalArgument(..))
+import Graphql.SelectionSet exposing (FragmentSelectionSet(..), SelectionSet(..))
+import Json.Decode as Decode
+
+
+type alias Fragments decodesTo =
+    { onSpecies : SelectionSet decodesTo EdgeCases.Interface.Species
```

Comment:
And there it is!  hurray!

Reply:
Okay so am I understanding this correctly that before your change, it wouldn't be aware that there `Character` could be a `Human` because `Human` implements multiple interfaces (`type Human implements Character & Species`)? So this `Fragment` would only have `{ onSpecies : ... }`, but no `onHuman`? Just trying to make sure I'm understanding what the core of this change is.

Thanks for working on this!

Reply:
It's actually the opposite.  `Character.Fragment` would be missing the `onSpecies` fragment.  The code generator wasn't picking up that `Species` interface implements `Character` interface.

This was caused by the generator code only looking at the `possibleTypes` field of an Interface which only lists Objects, not other interfaces (which actually seems like a bug in Graphql schema introspection).  So it would see that `Human` (an object) is a `possibleType` of `Character` but it wouldn't see `Species` (an interface).